platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1134,4 +1138,39 @@ String getFontFallbackFilePath(int importance, int weightHint) {
         return fontFileParser.getFontFallback(importance, weightHint);
     }
 
+    // Tmp Asset Path
+    // ==============
+    String copyAssetToTmpPath(String path) {
+
+        String tmpDir = mapView.getContext().getExternalCacheDir().getAbsolutePath();
+
+        try {
+            InputStream in = assetManager.open(path);
+            File outFile = new File(tmpDir, path);
+
+            if (!outFile.exists()) {
+                OutputStream out = new FileOutputStream(outFile);
+
+                // Copy File
+                byte[] buffer = new byte[1024];
+                int read;
+                while ((read = in.read(buffer)) != -1) {
+                    out.write(buffer, 0, read);
+                }
+
+                in.close();
+                out.flush();
+                out.close();
+            } else {
+                in.close();
+            }
+
+            return outFile.getAbsolutePath();
+        } catch (IOException e) {
```

Comment:
It's more of a Java thing, but streams should be close in a `finally` block in general.